patch for the latest dlrm

dkorchevgithub opened the pull request.

egkiastas committed:
Update run_local.sh

egkiastas committed:
Update run_and_time.sh

guschmue committed:
update readme for r0.7 branch

guschmue committed:
create r0.7 branch

nvpohanh committed:
Fix submission checker error for SingleStream performance logs

- LoadGen prints out `Single Stream`, but the checker failed at getting min_query_count
because of unexpected scenario name.
- Fixed several spelling errors.
- Prints out exception detail when there is an exception instead of silently failing.
- Fixed min_query_count requirement for Server scenario since 99th percentile is used.

christ1ne committed:
Merge pull request #720 from nvpohanh/dev-nvpohanh-checker-singlestream-error

Fix submission checker error for SingleStream performance logs

mnaumovfb committed:
Moving onnx runtime that was somehow left in the old dircetory.

mnaumovfb committed:
Enabling support for variable batch size.

ephrem-git committed:
added availability check

nvpohanh committed:
Fix truncate_accuracy_log.py failure on 3DUnet accuracy logs

mnaumovfb committed:
Adjusting import and location of user.conf and mlperf.conf. Also, removing some extra prints.

georgelyuan committed:
Fixing TEST01 loadgen logic to account for min_sample_count and min_query_count

georgelyuan committed:
removing debug cout

georgelyuan committed:
cleaning up logic

guschmue committed:
Merge pull request #726 from nvpohanh/dev-nvpohanh-open-utf8

Fix truncate_accuracy_log.py failure on 3DUnet accuracy logs

guschmue committed:
allow all scenarios in open, make Server optional

guschmue committed:
add more fields to summary report

georgelyuan committed:
explicitly point out which accuracy script to call for part III

nv-rborkar committed:
Add features to review compliance logs in branch r0.7

nv-rborkar committed:
Fix test04-b documentation

christ1ne committed:
Merge pull request #732 from guschmue/gs/server_is_opt

allow all scenarios in open, make Server optional

nv-rborkar committed:
verify_performance.txt is not necessary for TEST04-B

christ1ne committed:
Merge pull request #743 from nvpohanh/dev-pohanh-r0.7-merge

merge master to r0.7

Chujingjun committed:
allow custom model name in open division

Chujingjun committed:
add necessary double quotes for csv

guschmue committed:
Merge pull request #747 from Chujingjun/alvin_custom_model_name

allow custom model name in open division

guschmue committed:
few fixes to checker to deal with the full submission

guschmue committed:
workarounds for v0.7 submission

guschmue committed:
Merge branch 'r0.7' into r0.7

guschmue committed:
Merge pull request #735 from nv-rborkar/r0.7

Duplicate: Submission checker should review compliance test logs

nv-rborkar committed:
Patch to ignore extraneous files in compliance dir

guschmue committed:
Merge pull request #751 from nv-rborkar/r0.7

Patch to ignore extraneous files in compliance dir

guschmue committed:
Merge pull request #752 from mnaumovfb/master

DLRM: TensorFlow backend initial implementation

guschmue committed:
support for infered offline results

guschmue committed:
change the way we infer for offline, add final report generation

guschmue committed:
some fixes to final results spreadsheet

guschmue committed:
use quote in csv for system name, dont use 0 for accelorators

guschmue committed:
split results in closed/open, datacenter/edge

ens-lg4 committed:
allow optional truncation of the squad dataset, to make faster accuracy runs (full by default)

ens-lg4 committed:
no need to change into language/bert directory for an --accuracy run

ens-lg4 committed:
using the local version, which has the same functionality, but fewer dependencies

guschmue committed:
Merge pull request #756 from dividiti/dvdt-bert

More flexibility to simplify working with BERT outside of Docker

guschmue committed:
Merge pull request #737 from nv-rborkar/r0.7_test4_doc

Fix test04-b documentation

guschmue committed:
Merge pull request #724 from ephrem-git/submission_checker_br

added availability check

psyhtest committed:
Use tf.compat.v1 for compatibility with v2.

thaink committed:
Add EndTest function to loadgen

thaink committed:
change name to AbortTest

thaink committed:
Merge pull request #770 from thaink/add_end_test

Add EndTest function to loadgen

guschmue committed:
merge r0.7 to master

guschmue committed:
Merge pull request #780 from mlcommons/gs/merge_r0.7_to_master

merge r0.7 to master

Commit:
Create bert-calibration.txt

guschmue committed:
Merge pull request #758 from dividiti/dvdt-tf.compat.v1

[BERT] Make compatible with both TF v1 and v2 by using tf.compat.v1

Commit:
Update bert-calibration.txt

one per row

Commit:
Create README.md

add some explanations

Commit:
Merge pull request #781 from mlcommons/bert-calibration

Create bert-calibration.txt

sub-mod committed:
Create cla.yml

guschmue committed:
Merge pull request #785 from sub-mod/master

Create cla.yml for cla bot

guschmue committed:
Update README.md

sub-mod committed:
Update cla.yml

sub-mod committed:
Update cla.yml

sub-mod committed:
Update cla.yml

guschmue committed:
Merge branch 'master' into guschmue-patch-1

guschmue committed:
Merge pull request #786 from mlcommons/guschmue-patch-1

Update README.md

sub-mod committed:
Update cla.yml

nvpohanh committed:
Update upscale_coco.py (#771)

Fixes "file extension change" that was missing (and affecting those following README instructions)

See also #569.

psyhtest committed:
BERT: optional max_examples limit in accuracy mode. (#799)

dependabot[bot] committed:
Bump tensorflow from 1.12.0 to 2.3.1 in /others/cloud/language_modeling (#769)

Bumps [tensorflow](https://github.com/tensorflow/tensorflow) from 1.12.0 to 2.3.1.
- [Release notes](https://github.com/tensorflow/tensorflow/releases)
- [Changelog](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md)
- [Commits](https://github.com/tensorflow/tensorflow/compare/v1.12.0...v2.3.1)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

nvpohanh committed:
Logging Format Change (#779)

* Logging Format Change: step 1 - switch detailed log to new logging format

* More loadgen changes for new logging format, submission checker change, and new log parser

* Remove fallthrough warning

* toggle default strictness in log parser

* Fix submission checker errors

* Remove quotes from summary logs

dependabot[bot] committed:
Bump pillow from 6.2.0 to 7.1.0 in /others/cloud/single_stage_detector (#803)

Bumps [pillow](https://github.com/python-pillow/Pillow) from 6.2.0 to 7.1.0.
- [Release notes](https://github.com/python-pillow/Pillow/releases)
- [Changelog](https://github.com/python-pillow/Pillow/blob/master/CHANGES.rst)
- [Commits](https://github.com/python-pillow/Pillow/compare/6.2.0...7.1.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump tensorflow from 2.3.1 to 2.4.0 in /others/cloud/language_modeling (#804)

Bumps [tensorflow](https://github.com/tensorflow/tensorflow) from 2.3.1 to 2.4.0.
- [Release notes](https://github.com/tensorflow/tensorflow/releases)
- [Changelog](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md)
- [Commits](https://github.com/tensorflow/tensorflow/compare/v2.3.1...v2.4.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

sub-mod committed:
Update cla.yml

guschmue committed:
remove obsolete code in others (#816)

nvpohanh committed:
Fix compliance scripts to match new logging format (#818)

tjablin committed:
10 minutes is the new minimal duration (#820)

sub-mod committed:
Add MLCOMMONS_BOT_CLA_TOKEN

sub-mod committed:
Update cla.yml

keithachorn-intel committed:
SubmissionExample to ResNet-50 and bugfixes (#819)

* Updated SubmissionExample to ResNet-50 and resolved bugfixes

* Scrubbed references to MobileNet

* Clarified repo name, not suite name.

* Included reference output.

nvpohanh committed:
Update submission-checker.py for new 600 secs requirement (#825)

Update submission-checker.py for new 600 secs requirement

Also update the number of perf runs for Server scenario from 5 to 1

nvpohanh committed:
Fix error when inferring Offline QPS from SingleStream logs (#829)

Commit:
add missing library for windows (#831)

psyhtest committed:
Update LoadGen version to v1.0. (#834)

tjablin committed:
Fix link (#826)

Co-authored-by: christ1ne <christine.cheng@intel.com>

nvpohanh committed:
Allow a system to be in Datacenter and Edge simultaneously (#838)

* Allow a system to be in Datacenter and Edge simultaneously

* More fix to handle datacenter&edge system correctly in submission checker

TheKanter committed:
fix typo (#840)

someone mis-spelled interfaces and i'm fixing it.

guschmue committed:
Merge remote-tracking branch 'origin/r1.0'

guschmue committed:
merge some changes from 1.0 branch back to master

guschmue committed:
merge fix for 841

guschmue committed:
fix typo

tjablin committed:
Add missing return statement (#839)

Broken by #779.

guschmue committed:
merge r1.0 back to master

tjablin committed:
Add missing letters (#843)

Co-authored-by: christ1ne <christine.cheng@intel.com>

georgelyuan committed:
submission packing script for https://github.com/mlcommons/policies/issues/61 (#846)

tjablin committed:
Update Seeds to 1.0 (#847)

guschmue committed:
fix docker images (#844)

Co-authored-by: christ1ne <christine.cheng@intel.com>

guschmue committed:
updates seeds in submission-checker (#852)

cesar-avalos3 committed:
Change the way bert/run.py launches the accuracy-squad script by accounting the absolute path (#848)

Co-authored-by: Guenther Schmuelling <guschmue@microsoft.com>

tjablin committed:
Add flags and annotations links (#855)

Everyone seems to use the  --use-inv-map flag. It seems cruel to force everyone to discover it on their own.

Co-authored-by: Guenther Schmuelling <guschmue@microsoft.com>

nv-rborkar committed:
Add optional WAR for test04-B - offline test breaks contiguous memory requirements to stress SUT (#845)

Co-authored-by: Guenther Schmuelling <guschmue@microsoft.com>

corehalt committed:
Add PyTorch ImageNet pre-processing (#861)

dkorchevgithub committed:
patch for the latest dlrm

corehalt committed:
New PyTorch ResNet50-v1.5 links (#876)

* New PyTorch ResNet50-v1.5 links

* Delete unused import

tjablin committed:
Guard max_latency_ with latencies_mutex_ (#878)

* Guard max_latency_ with latencies_mutex_

There was a race in the computation of atomic max and std::memory_order_release
is not a valid order for atomic loads. The code isn't performance critical, so
it is easier and safer to use mutexes instead.

* Initialize max_latency_

Co-authored-by: Guenther Schmuelling <guschmue@microsoft.com>

Commit:
more intuitive error message (#883)

we can just update the master for now

tjablin committed:
Merge 1.0 (#910)

* pull changes from master to r1.0 (#851)

* fix typo

* Add missing return statement (#839)

Broken by #779.

* Add missing letters (#843)

Co-authored-by: christ1ne <christine.cheng@intel.com>

* submission packing script for https://github.com/mlcommons/policies/issues/61 (#846)

* Update Seeds to 1.0 (#847)

* fix docker images (#844)

Co-authored-by: christ1ne <christine.cheng@intel.com>

Co-authored-by: Guenther Schmuelling <guschmue@microsoft.com>
Co-authored-by: tjablin <tjablin@google.com>
Co-authored-by: georgelyuan <53881988+georgelyuan@users.noreply.github.com>

* Handle NaN/Inf correctly in LoadGen Logger (#837)

* Handle NaN/Inf correctly in LoadGen Logger

* Initialize target_latency_percentile in Offline scenario to avoid NaNs

Co-authored-by: Guenther Schmuelling <guschmue@microsoft.com>

* Add system description JSON fields for power submissions (#853)

* Check and parse power logs in submission checker (#854)

* Check and parse power logs in submission checker

* fixup required_perf_files in submission checker

Co-authored-by: Guenther Schmuelling <guschmue@microsoft.com>

* Print out correct error message when seeds are wrong (#858)

* take out the master content (#859)

Co-authored-by: Guenther Schmuelling <guschmue@microsoft.com>

* Fix links to benchmark and update submission date (#868)

* Fix message when performance_sample_count is too small (#865)

Co-authored-by: Guenther Schmuelling <guschmue@microsoft.com>

* Fix SingleStream latency metric in submission checker (#872)

* Add Power WG's check.py checking to submission checker (#873)

* Add Power WG's check.py checking to submission checker

* Update power-dev submodule to latest r1.0

Co-authored-by: Guenther Schmuelling <guschmue@microsoft.com>

* Update Power WG check.py to latest r1.0 (#884)

* ignore .github directory (#881)

* Change power metric to J/inference for SingleStream/MultiStream (#886)

* Change power metric to J/inference for SingleStream/MultiStream

* fix a spelling typo

* Fix power metric for SingleStream and update power-dev hash

* Fix typo

* fix url in results report (#890)

* Handle timezone differences when parsing power samples (#896)

* use host_processors_per_node in final report (#897)

* use host_processors_per_node in final report

* report per node processors

* fix typo

* New PyTorch ResNet50-v1.5 links (#876)

* New PyTorch ResNet50-v1.5 links

* Delete unused import

* Guard max_latency_ with latencies_mutex_ (#878)

* Guard max_latency_ with latencies_mutex_

There was a race in the computation of atomic max and std::memory_order_release
is not a valid order for atomic loads. The code isn't performance critical, so
it is easier and safer to use mutexes instead.

* Initialize max_latency_

Co-authored-by: Guenther Schmuelling <guschmue@microsoft.com>

* more intuitive error message (#883)

we can just update the master for now

Co-authored-by: christ1ne <christine.cheng@intel.com>
Co-authored-by: Guenther Schmuelling <guschmue@microsoft.com>
Co-authored-by: georgelyuan <53881988+georgelyuan@users.noreply.github.com>
Co-authored-by: nvpohanh <53919306+nvpohanh@users.noreply.github.com>
Co-authored-by: kahmed10 <15948690+kahmed10@users.noreply.github.com>
Co-authored-by: Core.Halt <46298335+corehalt@users.noreply.github.com>

Commit:
add submission runner files (#912)

DZFromArm committed:
Update backend_tf.py (#913)

Add thread count configuration and optimize_for_inference into the Tensorflow backend.

TheKanter committed:
TM (#916)

inserting trademark symbol

gfursin committed:
changing github.com/mlperf to github.com/mlcommons in a few places (otherwise git clone doesn't work) (#919)

guschmue committed:
add opset-11 onnx models, depreciated mobilenet (#921)

dkorchevgithub committed:
updated Docker CPU as suggested in

Issues:
https://github.com/mlcommons/inference/issues/917
https://github.com/mlcommons/inference/issues/604

dkorchevgithub committed:
updated for issues  #917 and #604

dkorchevgithub committed:
Update README.md

Drizshko committed:
Added comments (#925)

cfRod commented on recommendation/dlrm/pytorch/docker_gpu/Dockerfile:
```diff
@@ -38,28 +38,20 @@ RUN pip install --upgrade pip && \
     ldconfig
 
 RUN conda install pytorch torchvision -c pytorch
-RUN pip install tensorflow-gpu==1.14 onnxruntime-gpu
```
Same comment as before: does this reference code depend on tensorflow  (since this a reference implementation for pytorch)?

cfRod commented on recommendation/dlrm/pytorch/docker_cpu/Dockerfile:
```diff
@@ -31,7 +31,7 @@ RUN cd /opt && \
-RUN conda install pytorch-cpu torchvision-cpu -c pytorch
+RUN conda install pytorch torchvision cpuonly -c pytorch
 RUN pip install future pillow onnx opencv-python-headless tensorflow onnxruntime
 RUN pip install Cython && pip install pycocotools
```
Does this reference code depend on tensorflow (since this a reference implementation for pytorch)?

dkorchevgithub replied:
Could you use PR code from https://github.com/dkorchevgithub/inference/tree/master/recommendation

The container has tensorflow in it, but we will need to do more testing with it. The scripts provided are for pytorch at this point. Please feel free to test it and let us know if you find new issues. Thank you for your feedback.

dkorchevgithub committed:
updating readme

psyhtest commented on recommendation/dlrm/pytorch/README.md:
```diff
@@ -37,34 +37,35 @@ pip install pydot
 
 ### Prepare the code and dataset
-1. Download or clone the MLPerf [inference](https://github.com/mlperf/inference) and [trainining](https://github.com/mlperf/training) code
+1. Download or clone the MLCommons [inference](https://github.com/mlcommons/inference) and [trainining](https://github.com/mlcommons/training) code
```
trainining ->  training

cfRod commented on recommendation/dlrm/pytorch/README.md:
```diff
@@ -1,12 +1,12 @@
 | name | framework | acc. | AUC | dataset | weights  | size | prec. | notes |
 | ---- | ---- | ---- | ---- | ---- | ---- | ---- | ---- | ---- |
-| dlrm (debugging) | PyTorch | 78.82% | N/A | [Criteo KaggleDAC](https://labs.criteo.com/2014/02/kaggle-display-advertising-challenge-dataset/)       | N/A                                                                     | ~1GB | fp32 |                          |
+| dlrm (debugging) | PyTorch | 78.82% | N/A | [Criteo KaggleDAC (not active)](https://labs.criteo.com/2014/02/kaggle-display-advertising-challenge-dataset/)       | N/A                                                                     | ~1GB | fp32 |                          |
```
This link is not active. Is there an alternative that will be provided?

guschmue committed:
[WIP] remove chromium build system support (#929)

* only support cmake and python for loadgen build

* updated version number

guschmue committed:
tool to analyze the loadgen trace (#945)

dkorchevgithub committed:
fixed typo noticed by @psyhtest

dkorchevgithub committed:
Merge branch 'master' into master

LukeIreland1 committed:
[BERT] Decoupled TensorFlow from PyTorch and removed unused imports (#951)

guschmue committed:
fix for 933 (#940)

* fix for 933

* fix submodules

* fix submodules

Co-authored-by: rameshchukka <rnaidu02@yahoo.com>

nvitramble committed:
Fix typos in API bindings (#953)

Co-authored-by: Guenther Schmuelling <guschmue@microsoft.com>

nvpohanh committed:
Update power-dev submodule to latest master (#955)

nvitramble committed:
Add quantized BERT PyTorch reference model (#937)

Co-authored-by: Guenther Schmuelling <guschmue@microsoft.com>

guschmue committed:
add support for tflite wheel (#958)

ykurlaev committed:
Use UTC timestamps for power (#898)

Co-authored-by: Guenther Schmuelling <guschmue@microsoft.com>

guschmue committed:
remove absl dependency from setup.py, fix versions (#960)

nvitramble committed:
Add end on device support to Loadgen (#952)

* Add end on device callback to QuerySamplesComplete()

* Add end on device callback to QuerySamplesComplete()

* Fix submodule update

Co-authored-by: Guenther Schmuelling <guschmue@microsoft.com>

rnaidu02 committed:
Update README.md (#961)

Co-authored-by: Guenther Schmuelling <guschmue@microsoft.com>

psyhtest committed:
Merge branch 'master' into master

dkorchevgithub committed:
Updated Readme for DLRM and Docker files (#950)

* patch for the latest dlrm

* updated Docker CPU as suggested in

Issues:
https://github.com/mlcommons/inference/issues/917
https://github.com/mlcommons/inference/issues/604

* updated for issues  #917 and #604

* Update README.md

* updating readme

* fixed typo noticed by @psyhtest

Co-authored-by: Anton Lokhmotov <psyhtest@users.noreply.github.com>

dkorchevgithub committed:
Merge pull request #12 from mlcommons/master

Updated Readme for DLRM and Docker files (#950)

nv-jinhosuh committed:
3D UNet performing KiTS19 kidney tumor task (#935)

* Rename 3d-unet to 3d-unet-brats19

* Bringup 3d-unet-kits19, 3DUNet performing KiTS19 kidney/tumor task

Please refer to README.md for details

* preprocessing reshapes original KiTS19 imaging/segmentations
  into common voxel spacing; also reshapes them so that they
  become friendly to sub-volume inferencing

* timed portion, i.e. portion that uses loadgen for fetching reqs
  and returning results, is 1) splitting preprocessed image, 2) do
  sub-volume inferencing as many as needed, 3) stitching these
  inference results upon sub-volumes, and 4) do ArgMax to return
  segmentation prediction results in the same shape of preprocessed
  image

* Update nnUnet submodule for 3d-unet-brats19

* Addressing LGTM warnings

* JSON dumps prettier

* Removing unnecessary comment

* Remove unsupported scenario

dkorchevgithub committed:
adding lines suggested by @EtoDemerzel0427 to fix "good" count

dkorchevgithub committed:
Merge branch 'master' of https://github.com/dkorchevgithub/inference

dkorchevgithub committed:
Merge branch 'master' into master

guschmue committed:
branch r1.1

nvpohanh committed:
Update power-dev submodule to latest master 5e3dbea (#967)

Co-authored-by: Guenther Schmuelling <guschmue@microsoft.com>

rnaidu02 committed:
Merge branch 'master' into master
